fog/ledger/server/src/router_service.rs

```diff
@@ -80,3 +84,94 @@ where
         });
     }
 }
+
+/// Used for the implementation of FogKeyImageApi::check_key_images(),
+/// the legacy unary key-image API, for LedgerRouterService.
+async fn unary_check_key_image_inner<E>(
+    request: Message,
+    query_retries: usize,
+    enclave: E,
+    sink: UnarySink<Message>,
+    shard_clients: Vec<Arc<KeyImageStoreApiClient>>,
+    scope_logger: Logger,
+) -> Result<(), grpcio::Error>
+where
+    E: LedgerEnclaveProxy,
+{
+    let result = handle_query_request(
+        request,
+        enclave,
+        shard_clients,
+        query_retries,
+        scope_logger.clone(),
+    )
+    .await;
+
+    match result {
+        Ok(mut response) => {
+            if response.has_check_key_image_response() {
+                sink.success(response.take_check_key_image_response()).await
+            } else {
+                let error = rpc_internal_error(
+                    "Inavlid LedgerRequest response",
+                    "No check key image response to client's key image request.".to_string(),
```

Comment:
Suggest renaming to something like "Cannot provide a check key image response to the client's key image request."

Reply:
Sounds good.